src/lib/Microsoft.Health.Events/EventHubProcessor/AssignedPartitionProcessor.cs

```diff
@@ -0,0 +1,231 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Messaging.EventHubs;
+using Azure.Messaging.EventHubs.Consumer;
+using Azure.Messaging.EventHubs.Primitives;
+using Azure.Messaging.EventHubs.Processor;
+using Microsoft.Health.Events.EventCheckpointing;
+using Microsoft.Health.Events.EventConsumers.Service;
+using Microsoft.Health.Events.Model;
+using Microsoft.Health.Logging.Telemetry;
+
+namespace Microsoft.Health.Events.EventHubProcessor
+{
+    // This class was modeled after a sample Event Hub custom processor here:
+    // https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample08_CustomEventProcessor.md
+    public class AssignedPartitionProcessor : PluggableCheckpointStoreEventProcessor<EventProcessorPartition>
+    {
+        private IEventConsumerService _eventConsumerService;
+
+        private ICheckpointClient _checkpointClient;
+
+        private ITelemetryLogger _logger;
+
+        private readonly string[] _assignedPartitions;
+
+        public AssignedPartitionProcessor(
+            IEventConsumerService eventConsumerService,
+            StorageCheckpointClient checkpointClient,
+            ITelemetryLogger logger,
+            string[] assignedPartitions,
+            int eventBatchMaximumCount,
+            string consumerGroup,
+            TokenCredential tokenCredential,
+            string eventHubName,
+            string fullyQualifiedNamespace,
+            EventProcessorOptions clientOptions = default)
```

Comment:
Should we set the `EventProcessorOptions.MaximumWaitTime` to the value we define in `EventBatchingOptions.FlushTimespan`?